packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.ts

```diff
@@ -1,102 +1,260 @@
+    /**
+     * Flat values like string or number should only be added in `base` styles.
+     *
+     * E.g. if you pass `display="block"`, it should only put that style in base style and not in media queries
+     * ```js
+     * // Output should be just this-
+     * display: block;
+     *
+     * // And not this-
+     * display: block;
+     * media (min-width: s) {
+     *   display: block;
+     * }
+     *
+     * media (min-width: m) {
+     *   display: block
+     * }
+     * //  and more ...
+     * ```
+     */
+    if (size === 'base') {
+      return value;
+    }
+
+    return undefined;
+  }
+
+  if (isEmpty(value)) {
+    return undefined;
+  }
+
+  if (isReactNative()) {
+    // In React Native, we map the value `s` token on priority (since the breakpoint maps to mobiles in useBreakpoint hook).
+    // We further look into smaller sizes, then we check base size.
+    // Then we return the first non-undefined value in this priority
+    const priorityArray = [value.s, value.xs, value.base];
+    return priorityArray.find((val) => val !== undefined);
+  }
+
+  return value[size];
+};
+
+const getSpacingValue = (
+  spacingValue:
+    | MakeValueResponsive<SpacingValueType | ArrayOfMaxLength4<SpacingValueType>>
+    | undefined,
+  theme: Theme,
+  size?: keyof Breakpoints,
+): string | undefined => {
+  if (isEmpty(spacingValue)) {
+    return undefined;
+  }
+
+  const responsiveSpacingValue: SpacingValueType | SpacingValueType[] = getResponsiveValue(
+    spacingValue as MakeValueResponsive<string | string[]>,
+    size,
+  );
+
+  if (isEmpty(responsiveSpacingValue)) {
+    return undefined;
+  }
+
+  if (responsiveSpacingValue === 'auto') {
+    return responsiveSpacingValue;
+  }
+
+  if (Array.isArray(responsiveSpacingValue)) {
+    return responsiveSpacingValue.map((value) => getSpacingValue(value, theme)).join(' ');
+  }
+
+  if (typeof responsiveSpacingValue === 'string' && responsiveSpacingValue.startsWith('spacing.')) {
+    const spacingReturnValue = getIn(theme, responsiveSpacingValue);
+    return isEmpty(spacingReturnValue) ? makeSpace(spacingReturnValue) : undefined;
+  }
+
+  // pixel or with unit values
+  return responsiveSpacingValue;
+};
+
+const getBackgroundValue = (
+  backgroundColor: BaseBoxProps['backgroundColor'],
+  theme: Theme,
+  size?: keyof Breakpoints,
+): string => {
+  const responsiveBackgroundValue = getResponsiveValue(backgroundColor, size);
+  const tokenValue = getIn(theme, `colors.${responsiveBackgroundValue}`);
+  return tokenValue ?? responsiveBackgroundValue;
+};
+
+const getBorderRadiusValue = (
+  borderRadius: BaseBoxProps['borderRadius'],
+  theme: Theme,
+  size?: keyof Breakpoints,
+): string | undefined => {
+  const responsiveBorderRadiusValue = getResponsiveValue(borderRadius, size);
+  return isEmpty(responsiveBorderRadiusValue)
+    ? undefined
+    : makeSize(getIn(theme, `border.radius.${responsiveBorderRadiusValue}`));
```

Comment:
We should use the `makeBorderSize` function here. 
```suggestion
    : makeBorderSize(getIn(theme, `border.radius.${responsiveBorderRadiusValue}`));
```

Reply:
ahaaa didn't know this existed. thanks!